jenkins-integration/dev/beaker/20_install_r10k.rb

```diff
@@ -0,0 +1,58 @@
+test_name "Install r10k"
+
+def install_r10k(host)
+  gem = '/opt/puppetlabs/puppet/bin/gem'
+  on(host, "#{gem} install r10k --no-document")
+end
+
+def create_r10k_config(host)
+  configdir = '/etc/puppetlabs/r10k'
+  on(host, "mkdir -p #{configdir}")
+  r10k_config = <<EOS
+# The location to use for storing cached Git repos
+:cachedir: '/opt/puppetlabs/r10k/cache'
+
+# A list of git repositories to create
+:sources:
+  # This will clone the git repository and instantiate an environment per
+  # branch in /etc/puppetlabs/code/environments
+  :puppetserver-perf-driver:
+    remote: 'git@github.com:puppetlabs/puppetlabs-puppetserver_perf_driver_dev_control'
+    basedir: '/etc/puppetlabs/code/environments'
+EOS
+  create_remote_file(host, "#{configdir}/r10k.yaml", r10k_config)
+end
+
+
+step "Install git" do
+  on(jenkins, puppet_resource("package git ensure=installed"))
+end
+
+step "Set up SSH key for github access" do
+  if !jenkins.file_exist?("/root/.ssh/id_rsa")
```

Comment:
Just FYI, Jenkins ends up using keys located at `/var/lib/jenkins/.ssh` because jobs execute as the `jenkins` user, and their home directory is configured to be the Jenkins installation root directory, `/var/lib/jenkins/`